reporting/api/src/main/java/org/jboss/windup/reporting/rules/CreateApplicationListReportRuleProvider.java

```diff
@@ -69,15 +77,51 @@ private void createIndexReport(GraphContext context)
         report.setReportFilename(OUTPUT_FILENAME);
 
         GraphService<WindupVertexListModel> listService = new GraphService<>(context, WindupVertexListModel.class);
-        WindupVertexListModel<ApplicationReportModel> applications = listService.create();
-        for (ApplicationReportModel applicationReportModel : applicationReportService.findAll())
+        WindupVertexListModel<ApplicationReportModel> appsListVertex = listService.create();
+
+
+        final Iterable<ApplicationReportModel> apps = applicationReportService.findAll();
+        List<ApplicationReportModel> appsList = new ArrayList();
+        for (ApplicationReportModel applicationReportModel : apps)
         {
             if (applicationReportModel.isMainApplicationReport() != null && applicationReportModel.isMainApplicationReport())
-                applications.addItem(applicationReportModel);
+                appsList.add(applicationReportModel);
         }
+        // Our current model doesn't keep the list order, but once I wrote, I'm leaving the sorting here for when it does.
+        appsList.sort(new AppRootFileNameComparator());
```

Comment:
This is Java 8 only method for `java.util.List`
Java 7 `java.util.List` doesn't have such method see https://docs.oracle.com/javase/8/docs/api/java/util/List.html#sort-java.util.Comparator- where it is described 

> since 1.8